src/intersection.js

```diff
@@ -23,5 +23,7 @@ var uniq = require('./uniq');
  *      R.intersection([1,2,3,4], [7,6,5,4,3]); //=> [4, 3]
  */
 module.exports = _curry2(function intersection(list1, list2) {
-  return uniq(_filter(flip(_contains)(list1), list2));
+  var lookupList = list1.length > list2.length ? list1 : list2;
+  var filteredList = lookupList == list1 ? list2 : list1;
```

Comment:
FWIW, I think that for Ramda, which will never use ES6 classes and all that jazz, "loose mode" is good enough (you can try that there). But are you worried about readability or performance? Because I think it's reasonable to expect consumers to read the pre-compilation source code.